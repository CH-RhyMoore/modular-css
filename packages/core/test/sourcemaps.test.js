"use strict";

var Consumer = require("source-map").SourceMapConsumer,
    
    namer = require("test-utils/namer.js"),
    
    Processor = require("../processor.js");

describe("/processor.js", () => {
    describe("sourcemap support", () => {
        var processor, map;
        
        beforeEach(() => {
            processor = new Processor({
                namer,
                map : {
                    inline : false
                }
            });
        });

        afterEach(() => {
            if(map && map.destroy) {
                map.destroy();
            }
        });

        it("should output correct single-file source maps", () =>
            processor.file(
                require.resolve("./specimens/simple.css")
            )
            .then(() => processor.output({
                to : "./packages/core/test/output/simple.css"
            }))
            .then((output) => {
                map = new Consumer(output.map.toString());

                expect(map.sources).toMatchSnapshot();

                const mappings = [];

                map.eachMapping((m) => mappings.push(m));

                expect(mappings).toMatchSnapshot();
            })
        );

        it("should output correct multi-file source maps", () =>
            processor.file(
                require.resolve("./specimens/local.css")
            )
            .then(() => processor.output({
                to : "./packages/core/test/output/local.css"
            }))
            .then((output) => {
                map = new Consumer(output.map.toString());

                expect(map.sources).toMatchSnapshot();

                const mappings = [];

                map.eachMapping((m) => mappings.push(m));

                expect(mappings).toMatchSnapshot();
            })
        );
    });
});
