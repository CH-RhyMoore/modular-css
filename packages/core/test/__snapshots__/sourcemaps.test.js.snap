// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/processor.js sourcemap support should output correct multi-file source maps 1`] = `
Array [
  "../specimens/folder/folder.css",
  "../specimens/local.css",
]
`;

exports[`/processor.js sourcemap support should output correct multi-file source maps 2`] = `
Array [
  Object {
    "generatedColumn": 0,
    "generatedLine": 1,
    "name": null,
    "originalColumn": 0,
    "originalLine": 1,
    "source": "../specimens/folder/folder.css",
  },
  Object {
    "generatedColumn": 52,
    "generatedLine": 1,
    "name": null,
    "originalColumn": 1,
    "originalLine": 1,
    "source": "../specimens/folder/folder.css",
  },
  Object {
    "generatedColumn": 0,
    "generatedLine": 2,
    "name": null,
    "originalColumn": 0,
    "originalLine": 2,
    "source": "../specimens/folder/folder.css",
  },
  Object {
    "generatedColumn": 10,
    "generatedLine": 2,
    "name": null,
    "originalColumn": 10,
    "originalLine": 2,
    "source": "../specimens/folder/folder.css",
  },
  Object {
    "generatedColumn": 22,
    "generatedLine": 2,
    "name": null,
    "originalColumn": 22,
    "originalLine": 2,
    "source": "../specimens/folder/folder.css",
  },
  Object {
    "generatedColumn": 24,
    "generatedLine": 2,
    "name": null,
    "originalColumn": 24,
    "originalLine": 2,
    "source": "../specimens/folder/folder.css",
  },
  Object {
    "generatedColumn": 0,
    "generatedLine": 3,
    "name": null,
    "originalColumn": 0,
    "originalLine": 1,
    "source": "../specimens/local.css",
  },
  Object {
    "generatedColumn": 44,
    "generatedLine": 3,
    "name": null,
    "originalColumn": 1,
    "originalLine": 1,
    "source": "../specimens/local.css",
  },
  Object {
    "generatedColumn": 0,
    "generatedLine": 4,
    "name": null,
    "originalColumn": 0,
    "originalLine": 5,
    "source": "../specimens/local.css",
  },
  Object {
    "generatedColumn": 9,
    "generatedLine": 4,
    "name": null,
    "originalColumn": 9,
    "originalLine": 5,
    "source": "../specimens/local.css",
  },
  Object {
    "generatedColumn": 27,
    "generatedLine": 4,
    "name": null,
    "originalColumn": 27,
    "originalLine": 5,
    "source": "../specimens/local.css",
  },
  Object {
    "generatedColumn": 29,
    "generatedLine": 4,
    "name": null,
    "originalColumn": 29,
    "originalLine": 5,
    "source": "../specimens/local.css",
  },
]
`;

exports[`/processor.js sourcemap support should output correct single-file source maps 1`] = `
Array [
  "../specimens/simple.css",
]
`;

exports[`/processor.js sourcemap support should output correct single-file source maps 2`] = `
Array [
  Object {
    "generatedColumn": 0,
    "generatedLine": 1,
    "name": null,
    "originalColumn": 0,
    "originalLine": 1,
    "source": "../specimens/simple.css",
  },
  Object {
    "generatedColumn": 45,
    "generatedLine": 1,
    "name": null,
    "originalColumn": 1,
    "originalLine": 1,
    "source": "../specimens/simple.css",
  },
  Object {
    "generatedColumn": 0,
    "generatedLine": 2,
    "name": null,
    "originalColumn": 0,
    "originalLine": 1,
    "source": "../specimens/simple.css",
  },
  Object {
    "generatedColumn": 9,
    "generatedLine": 2,
    "name": null,
    "originalColumn": 9,
    "originalLine": 1,
    "source": "../specimens/simple.css",
  },
  Object {
    "generatedColumn": 20,
    "generatedLine": 2,
    "name": null,
    "originalColumn": 20,
    "originalLine": 1,
    "source": "../specimens/simple.css",
  },
  Object {
    "generatedColumn": 22,
    "generatedLine": 2,
    "name": null,
    "originalColumn": 22,
    "originalLine": 1,
    "source": "../specimens/simple.css",
  },
]
`;
